mlir/include/mlir/Bytecode/BytecodeImplementation.h

```diff
@@ -54,6 +54,10 @@ class DialectBytecodeReader {
   /// Retrieve the dialect version by name if available.
   virtual FailureOr<const DialectVersion *>
   getDialectVersion(StringRef dialectName) const = 0;
+  template <class T>
+  FailureOr<const DialectVersion *> getDialectVersion() {
+    return getDialectVersion(T::getDialectNamespace());
```

Comment:
why is there a non-const version of this one?

Reply:
Fixed :)